Marlin/src/lcd/menu/game/types.h

```diff
@@ -41,6 +49,130 @@ class MarlinGame {
+   */
+  static void set_color(const color color);
+
+  /**
+   * @brief Draw a horizontal line.
+   * @param x The x-coordinate of the start of the line.
+   * @param y The y-coordinate of the line.
+   * @param l The length of the line.
+   * @see https://github.com/olikraus/u8glib/wiki/userreference#drawhline
+   */
+  static void draw_hline(const game_dim_t x, const game_dim_t y, const game_dim_t l);
+
+  /**
+   * @brief Draw a vertical line.
+   * @param x The x-coordinate of the line.
+   * @param y The y-coordinate of the start of the line.
+   * @param l The length of the line.
+   * @see https://github.com/olikraus/u8glib/wiki/userreference#drawvline
+   */
+  static void draw_vline(const game_dim_t x, const game_dim_t y, const game_dim_t l);
+
+  /**
+   * @brief Draw a outlined rectangle (frame).
+   * @param x The x-coordinate of the top-left corner of the frame.
+   * @param y The y-coordinate of the top-left corner of the frame.
+   * @param w The width of the frame.
+   * @param h The height of the frame.
+   * @see https://github.com/olikraus/u8glib/wiki/userreference#drawframe
+   */
+  static void draw_frame(const game_dim_t x, const game_dim_t y, const game_dim_t w, const game_dim_t h);
+
+  /**
+   * @brief Draw a filled rectangle (box).
+   * @param x The x-coordinate of the top-left corner of the box.
+   * @param y The y-coordinate of the top-left corner of the box.
+   * @param w The width of the box.
+   * @param h The height of the box.
+   * @see https://github.com/olikraus/u8glib/wiki/userreference#drawbox
+   */
+  static void draw_box(const game_dim_t x, const game_dim_t y, const game_dim_t w, const game_dim_t h);
+
+  /**
+   * @brief Draw a pixel.
+   * @param x The x-coordinate of the pixel.
+   * @param y The y-coordinate of the pixel.
+   * @see https://github.com/olikraus/u8glib/wiki/userreference#drawpixel
+   */
+  static void draw_pixel(const game_dim_t x, const game_dim_t y);
+
+  /**
+   * @brief Draw a bitmap.
+   * @param x The x-coordinate of the top-left corner of the bitmap.
+   * @param y The y-coordinate of the top-left corner of the bitmap.
+   * @param bytes_per_row The number of bytes per row in the bitmap (Width = bytes_per_row * 8).
+   * @param rows The number of rows in the bitmap (= Height).
+   * @param bitmap The bitmap to draw.
+   * @see https://github.com/olikraus/u8glib/wiki/userreference#drawbitmap
+   */
+  static void draw_bitmap(const game_dim_t x, const game_dim_t y, const game_dim_t bytes_per_row, const game_dim_t rows, const pgm_bitmap_t bitmap);
+
+  /**
+   * @brief Draw a string.
+   * @param x The x-coordinate of the string.
+   * @param y The y-coordinate of the string.
+   * @param str The string to draw.
+   * @see lcd_moveto + lcd_put_u8str
+   * @note The font size is available using the GAME_FONT_WIDTH and GAME_FONT_ASCENT constants.
+   */
+  static int draw_string(const game_dim_t x, const game_dim_t y, const char *str);
```

Comment:
add comment to note that draw_string flushes on DWIN, so should be called after all other game elements are drawn only